### PR TITLE
warn the user before overwriting the existing file

### PR DESCRIPTION
**Pull request by AdarshdeepCheema**
Signed-off-by: Adarshdeep Cheema <adarshdeep.cheema@ibm.com>

## Proposed changes

This PR addresses Issue: https://github.com/zowe/zlux/issues/857 and https://github.com/zowe/zlux/issues/767

This PR depends upon the following PRs:

## Type of change
Please delete options that are not relevant.
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Change in a documentation
- [ ] Refactor the code 
- [ ] Chore, repository cleanup, updates the dependencies.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## PR Checklist
Please delete options that are not relevant.
- [ ] If the changes in this PR are meant for the next release / mainline, this PR targets the "staging" branch.
- [ ] My code follows the style guidelines of this project (see: [Contributing guideline](https://github.com/zowe/zlux/blob/master/CONTRIBUTING.md))
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] video or image is included if visual changes are made
- [ ] Relevant update to CHANGELOG.md
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works, or describe a test method below

### CODE REVIEW

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts [hunk 1]
```diff
@@ -425,12 +425,31 @@ export class MonacoService implements OnDestroy {
...
+                  obs.next('Cancel');
+                }
+              });
+            }, error => {
+              if(error.status == 404) {// if file does not exist at destination, then try to save it
+                if (result) {
+                  this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next('Save'));
+                }
+              } else{
+                this.snackBar.open(`Error: Failed to verify if "${result.directory + '/' + result.fileName}" already exists`,
```
Comment (DivergentEuropeans) [inline webClient/src/app/editor/code-editor/monaco/monaco.service.ts]: If metadata call fails, we should give the error status code

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts [hunk 2]
```diff
@@ -425,12 +425,31 @@ export class MonacoService implements OnDestroy {
...
           saveRef.afterClosed().subscribe(result => {
-          if (result) {
-            this.editorControl.saveBuffer(fileContext, result).subscribe(() => obs.next('Save'));
-          } else {
-            obs.next('Cancel');
-          }
-          
+            // Check if file already exists at destination
+            this.editorControl.getFileMetadata(result.directory + '/' + result.fileName).subscribe(r => {
+              const title = `"${result.fileName}" already exists. Do you want to replace it?`;
```
Comment (DivergentEuropeans) [inline webClient/src/app/editor/code-editor/monaco/monaco.service.ts:430]: In some areas, we don't use quotation marks for file name ie
```
this.snackBar.open(`${_activeFile.name} has been saved!`, 'Close', { duration: MessageDuration.Short, panelClass: 'center' });
```

```
this.snackBar.open(`${_activeDataset.name} could not be saved, feature not yet implemented.`, 
                           'Close', { duration: MessageDuration.Long,   panelClass: 'center' });   
```
so let's keep it consistent

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts [hunk 3]
```diff
@@ -425,12 +425,31 @@ export class MonacoService implements OnDestroy {
...
+              const warningMessage = 'A file or folder with the same name already exists in the folder monaco. Replacing it will overwrite its current contents..';
```
Comment (DivergentEuropeans) [inline webClient/src/app/editor/code-editor/monaco/monaco.service.ts]: First sentence isn't needed (covered by title) it should be

image

**Review by DivergentEuropeans — CHANGES_REQUESTED** [review]
Code doesn't look too shabby, just check out my comments. But I think UX could use a little work
Needs a little padding here (can be accomplished by hardcoding modal width like some places do)

image

Also please move the buttons to the bottom right, that appears to be the theme for most modals

image
image

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts (hunk 3 above)
Comment (AdarshdeepCheema) [reply webClient/src/app/editor/code-editor/monaco/monaco.service.ts]: Done

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts (hunk 2 above)
Comment (AdarshdeepCheema) [reply webClient/src/app/editor/code-editor/monaco/monaco.service.ts:430]: Done

File: webClient/src/app/editor/code-editor/monaco/monaco.service.ts (hunk 1 above)
Comment (AdarshdeepCheema) [reply webClient/src/app/editor/code-editor/monaco/monaco.service.ts]: Done

**Review by DivergentEuropeans — APPROVED** [review]

**Comment by DivergentEuropeans** [comment]
Works great thanks, changes addressed, merging 👍

**Pull request merged**